pytorch_lightning/utilities/parsing.py

```diff
@@ -177,60 +177,53 @@ def __repr__(self):
 def lightning_hasattr(model, attribute):
     """ Special hasattr for lightning. Checks for attribute in model namespace,
         the old hparams namespace/dict, and the datamodule. """
-    trainer = model.trainer
-
-    # Check if attribute in model
-    if hasattr(model, attribute):
-        attr = True
-    # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
-        if isinstance(model.hparams, dict):
-            attr = attribute in model.hparams
-        else:
-            attr = hasattr(model.hparams, attribute)
-    # Check if the attribute in datamodule (datamodule gets registered in Trainer)
-    elif trainer is not None and trainer.datamodule is not None and hasattr(trainer.datamodule, attribute):
-        attr = getattr(trainer.datamodule, attribute)
-    else:
-        attr = False
+    try:
+        lightning_getattr(model, attribute)
+    except AttributeError:
+        return False
 
-    return attr
+    return True
 
 
 def lightning_getattr(model, attribute):
     """ Special getattr for lightning. Checks for attribute in model namespace,
         the old hparams namespace/dict, and the datamodule. """
-    trainer = model.trainer
-
     # Check if attribute in model
     if hasattr(model, attribute):
-        attr = getattr(model, attribute)
+        return getattr(model, attribute)
+
     # Check if attribute in model.hparams, either namespace or dict
-    elif hasattr(model, 'hparams'):
+    if hasattr(model, 'hparams'):
         if isinstance(model.hparams, dict):
-            attr = model.hparams[attribute]
+            try:
+                return model.hparams[attribute]
+            except KeyError:
+                pass
```

Comment:
```suggestion
            model.hparams.get(attribute)
```

Reply:
Actually, I would like to leave it as is because we want `lightning_getattr` to continue to look for `attr` even when not found in `model.hparams`.

If we use the dictionary method `dict.get()`, I think it wouldn't be possible to see whether `model.hparams` has `attr` or not.